test/examples/change_streams.js

```diff
@@ -35,11 +36,20 @@ describe('examples(change-stream):', function() {
 
       // Start Changestream Example 1
       const collection = db.collection('inventory');
+
+      // process as a stream
       const changeStream = collection.watch();
-      const next = await changeStream.next();
+      changeStream.on('change', next => {
+        // process next document
+      });
+
+      // or as an iterator
+      const changeStreamIterator = collection.watch();
+      const next = await changeStreamIterator.next();
```

Comment:
I would drastically prefer to make the iterator variant the preferred way to consume a changeStream.

Reply:
The event emitter approach is more idiomatic of an "event-driven" "asynchronous" language/environment. I haven't heard an example of an unavoidable issue this would lead to, if you have rationale to not prefer the event emitter approach let's discuss here.

Reply:
a) EventEmitters frequently lead to memory leaks.
b) It is significantly more painful to handle errors in EventEmitter code than with Promises and async/await
c) In general the community is moving towards async/await for most things, so I think that is the better example to have up front.
d) Using the iterator approach allows a finer-grain approach to iterating over the changes in an object, by letting them ensure that a new change will not be consumed driver-side before the user calls `next`. The event-based API allows for the possibility of multiple change stream events being fired before an error is detected in user-land code, making determining the proper resume token complicated.